controllers/serviceclaim_controller.go

```diff
@@ -488,7 +488,42 @@ func (r *ServiceClaimReconciler) DeleteServiceBindingsAndSecret(
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *ServiceClaimReconciler) SetupWithManager(mgr ctrl.Manager) error {
+
+	genPred := predicate.GenerationChangedPredicate{}
+	reconcileOnRegisteredServiceUpdate := func(ctx context.Context, a client.Object) []reconcile.Request {
+		l := log.FromContext(ctx)
+		rs, ok := a.(*v1alpha1.RegisteredService)
+		if !ok {
+			l.Info("error parsing object to RegisteredService when mapping to ServiceClaim reconciliation trigger", "object", a)
+			return []reconcile.Request{}
+		}
+		if rs.Status.State != `Claimed` {
+			l.Info("Registered service is unclaimed, no service claim to reconcile", "registered-service", rs.Name)
+			return []reconcile.Request{}
+		}
+		serviceclaims := &v1alpha1.ServiceClaimList{}
+		opts := &client.ListOptions{}
+		if err := r.List(ctx, serviceclaims, opts); err != nil {
+			l.Error(err,
+				"unable to list the ServiceClaims and reconcile for Registered Service Updates",
+				"RegisteredService", rs.Name)
+			return []reconcile.Request{}
+		}
+		reply := make([]reconcile.Request, 0, 1)
+		for _, sc := range serviceclaims.Items {
+			if sc.Status.State == "Resolved" && sc.Status.RegisteredService == rs.Name {
+				reply = append(reply, reconcile.Request{NamespacedName: types.NamespacedName{
+					Namespace: sc.Namespace,
+					Name:      sc.Name,
+				}})
+				return reply
```

Comment:
this can be simplified, other than this LGTM
```suggestion
		for _, sc := range serviceclaims.Items {
			if sc.Status.State == "Resolved" && sc.Status.RegisteredService == rs.Name {
				return []reconcile.Request{{NamespacedName: types.NamespacedName{
					Namespace: sc.Namespace,
					Name:      sc.Name,
				}}})
```